src/panel/direction/RouteResult.jsx

```diff
@@ -73,7 +73,20 @@ export default class RouteResult extends React.Component {
     if (this.props.error) {
       return <div className="itinerary_no-result">
         <span className="icon-alert-triangle" />
-        <div>{_('Qwant Maps found no results for this itinerary.')}</div>
+        <div>{
+          this.props.error >= 500 && this.props.error < 600
+            ? _('The service is temporarily unavailable, please try again later.', 'direction')
+            : this.props.error === 422
+              ? _('Qwant Maps found no results for this itinerary.', 'direction')
+              : ''
```

Comment:
What if the `directionResponse` contains an empty list of routes ?   
As far as I understand, `this.props.error` will contain the response object, and the error message will no longer be visible.

Reply:
Here is the result
![image](https://user-images.githubusercontent.com/4726554/73853658-84648900-4831-11ea-8159-88e755f936a2.png)

With a React warning
```
Warning: Failed prop type: Invalid prop `error` of type `object` supplied to `RouteResult`, expected `number`.
    in RouteResult (created by DirectionPanel)
    in DirectionPanel (created by PanelManager)
    in div (created by PanelManager)
    in PanelManager
```

Reply:
I agree there is an inconsistency for empty results.

Maybe we can change returns from direction_api so error cases are easier to parse and we don't need to make assumptions such as `if (directionResponse && directionResponse.routes && directionResponse.routes.length > 0)` to separate valid results from errors in the Panel.

For example, always return an object, with an optional `error` field, and never return the error itself. Even if the API returns "correctly" but with an empty list of routes, fill the error property so it falls in a generic "no route found" case.

Reply:
I agree, I lacked a bit of global knowledge of this part of erdapfel to address this issue in the best possible way. I also thinked about including the optional error code in the same object that contains the data if there's no error... I'll try to make this the right way, and will let you check if it's ok :)